### PR TITLE
refactor: Make `error.kind` and `error.stack` non-optional in SDK errors

### PR DESCRIPTION
**Pull request by ncreated**
### What and why?

📦 This PR changes the `error.kind` (and also `error.stack`) in SDK error telemetry to be non-optional (always set).

This serves two purposes:
- adds more context to SDK errors and avoid having error logs with no `error.kind`,
- increases the `error.kind` cardinality for later aggregating SDK errors by their kind in "RUM Session Ended" metric.

### How?

The `error.kind` now defaults to the `#fileID`:
```
e.g. `DatadogSR/Recorder.swift`
```
This will increase the cardinality of `error.kind` in SDK errors telemetry.

The `error.stack` now defaults to `#fileID` and `#line`:
```
e.g. `DatadogSR/Recorder.swift:42
```
This will add more clarity on SDK errors origin in certain SDK version.

### Review checklist
- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [ ] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [ ] Run unit tests for Core, RUM, Trace, Logs, CR and WVT
- [ ] Run unit tests for Session Replay
- [ ] Run integration tests
- [ ] Run smoke tests
- [ ] Run tests for `tools/`

### CODE REVIEW

File: DatadogInternal/Tests/Telemetry/TelemetryTests.swift
Comment (ncreated) [inline DatadogInternal/Tests/Telemetry/TelemetryTests.swift:1]: 🎁 I did vast revamp of this test file. It now leverages most helpers added after it was first introduced:
- [`TelemetryMock`](https://github.com/DataDog/dd-sdk-ios/blob/876787a4b056350b8ace92501d53edcd910bd664/TestUtilities/Mocks/TelemetryMocks.swift#L55) to test `Telemetry` protocol extension;
- [Array helpers](https://github.com/DataDog/dd-sdk-ios/blob/876787a4b056350b8ace92501d53edcd910bd664/TestUtilities/Mocks/TelemetryMocks.swift#L86) for extracting telemetry messages;
- Simplified testing of the integration with mocked core - as telemetry is now covered in [integration unit tests](https://github.com/DataDog/dd-sdk-ios/blob/develop/Datadog/IntegrationUnitTests/Public/TelemetryCoreIntegrationTests.swift).

File: DatadogInternal/Tests/Telemetry/TelemetryTests.swift [hunk 1]
```diff
@@ -5,267 +5,179 @@
...
+        XCTAssertEqual(telemetry.messages.count, 2)
+        XCTAssertEqual(errors[0].message, #"SwiftError(description: "error description")"#)
+        XCTAssertEqual(errors[0].kind, "SwiftError")
+        XCTAssertEqual(errors[0].stack, #"SwiftError(description: "error description")"#)
+        XCTAssertEqual(errors[1].message, #"custom message - SwiftError(description: "error description")"#)
+        XCTAssertEqual(errors[1].kind, "SwiftError")
+        XCTAssertEqual(errors[1].stack, #"SwiftError(description: "error description")"#)
     }
 
-    func testTelemetryConfigurationMerge() {
```
Comment (ncreated) [inline DatadogInternal/Tests/Telemetry/TelemetryTests.swift:125]: [This test](https://github.com/DataDog/dd-sdk-ios/blob/876787a4b056350b8ace92501d53edcd910bd664/DatadogInternal/Tests/Telemetry/TelemetryTests.swift#L125-L139) was removed as it was not testing SDK logic. It was calling the mock and asserting if the mock is updated.

**Review by maciejburda — APPROVED** [review]
👌

File: DatadogInternal/Sources/Telemetry/Telemetry.swift [hunk 2]
```diff
@@ -173,21 +173,33 @@ extension Telemetry {
...
     ///   - file: The current file name.
     ///   - line: The line number in file.
-    ///   - file: The current file name.
-    ///   - line: The line number in file.
-    public func error(_ message: String, kind: String? = nil, stack: String? = nil, file: String = #file, line: Int = #line) {
-        error(id: "\(file):\(line):\(message)", message: message, kind: kind, stack: stack)
+    public func error(_ message: String, kind: String? = nil, stack: String? = nil, file: String = #fileID, line: Int = #line) {
+        func defaultKind(fileID: String) -> String {
+            // e.g. `DatadogInternal/Telemetry.swift` -> `DatadogInternal/TelemetryError`
+            return "\(fileID.hasSuffix(".swift") ? String(fileID.dropLast(6)) : fileID)Error"
```
Comment (maxep) [inline DatadogInternal/Sources/Telemetry/Telemetry.swift:191]: /question I don't see the benefit of doing this format, having `kind: "fileID"` and `stack: "\(fileID):\(line)"` will give us the same info and cardinality. We could avoid doing string parsing/formatting, unless I'm missing something?

**Review by maxep — COMMENTED** [review]
Looks good, thanks for the revamp 👍 
I have left a question regrading the `kind` format.

File: DatadogInternal/Sources/Telemetry/Telemetry.swift (hunk 2 above)
Comment (ncreated) [reply DatadogInternal/Sources/Telemetry/Telemetry.swift]: Agree, it isn't necessary 👍. ✅ Changed to `kind: #fileID`.

**Review by maxep — APPROVED** [review]
Thank you :)

**Review by maciejburda — APPROVED** [review]
👌

**Pull request merged**